pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -73,7 +73,7 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 		serverVersion: sv,
 		reconcileIn:   time.Second * 5,
 		crons:         NewCronRegistry(),
-		statusMutex:   new(sync.Mutex),
```

Comment:
if we remove Mutex, I don't this that operator will work if we run it with multiple reconcilers in parallel, like this
```
--- pkg/controller/perconaservermongodb/psmdb_controller.go
+++ pkg/controller/perconaservermongodb/psmdb_controller.go
@@ -82,7 +82,10 @@ func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
        // Create a new controller
-       c, err := controller.New("psmdb-controller", mgr, controller.Options{Reconciler: r})
+       c, err := controller.New("psmdb-controller", mgr, controller.Options{
+               Reconciler:              r,
+               MaxConcurrentReconciles: 5,
+       })
        if err != nil {
                return err
        }
```

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/472/commits/10cc0ffb4848de9f64c8d5762d5e6d684e6a1dac

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/472/commits/10cc0ffb4848de9f64c8d5762d5e6d684e6a1dac